../usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88152 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=14, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59620)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=15, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59622)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=16, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59624)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=17, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59626)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=18, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59628)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=19, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59630)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=20, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59632)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=21, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59634)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/requests/structures.py:41: ResourceWarning: unclosed <socket.socket fd=22, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 59636)>
  self._store = OrderedDict()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=9>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=10>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=11>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=12>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=13>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=14>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=15>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=16>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=17>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=18>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=19>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=20>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=21>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=22>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88183 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=23>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
../usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88254 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=25>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=28>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88291 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
../usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=35>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88356 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88202 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88325 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=24>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=26>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=27>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=29>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=30>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=31>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=32>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=33>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=34>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88375 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88479 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88591 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88698 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88701 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88705 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88708 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 88711 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback

----------------------------------------------------------------------
Ran 14 tests in 19.736s

OK
Setting up test:
	thread: 3
	queue size: 7
	policy: dh
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 14
Setting up test:
	thread: 3
	queue size: 7
	policy: dh
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 1
Setting up test:
	thread: 3
	queue size: 16
	policy: random
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 16
Requests failed: 8
Setting up test:
	thread: 3
	queue size: 16
	policy: random
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 16
Requests failed: 0
Setting up test:
	thread: 3
	queue size: 16
	policy: random
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 13
Requests failed: 4
Setting up test:
	thread: 3
	queue size: 7
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 14
Setting up test:
	thread: 3
	queue size: 7
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 1
Setting up test:
	thread: 3
	queue size: 80
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 80
Requests failed: 0
Setting up test:
	thread: 9
	queue size: 80
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 80
Requests failed: 0
Setting up test:
	thread: 3
	queue size: 80
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 80
Requests failed: 0
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
